### ISSUE TITLE
fine-tuning a llamas model with culanes data

### ISSUE THREAD

**Issue by MehdiAlilou** [voldemortX/pytorch-auto-drive#108]
I have a model trained on llamas dataset using erfnet_scnn_llamas configuration and would like to introduce culanes data set to this model as well. I use --checkpoint to load the model and used culanes config file for the second round training on culanes. However I get an error because flattened_size=4500, for erfnet_culanes.py config file  and it is 4400 in the erfnet_llamas.py. This causes an error while loading the checkpoint. I tried to change the input size for culane config file and change the model parameters as well but still getting error. Could you please guide me how to solve this issue?

**Comment by MehdiAlilou** [comment]
I used the default config file available in /configs/lane_detection/scnn/erfnet_culane.py. The input size is different from the llamas' config file. I tried to make the same input size as the llamas' config file but it did not work.
Previously I used erfnet_llamas.py to train on llamas.

**Comment by voldemortX** [comment]
ok. Let me check that when I have free GPUs later today.

**Comment by MehdiAlilou** [comment]
Thank you very much.

**Comment by voldemortX** [comment]
@MehdiAlilou I can't use my testing env for now. But I think this config could do the finetuning, maybe try it out?
[culane_finetune.zip](https://github.com/voldemortX/pytorch-auto-drive/files/9069097/culane_finetune.zip)

**Comment by MehdiAlilou** [comment]
@voldemortX thanks for your prompt response. I will test it tomorrow and will let you know the result. By the way I had another question while using --train mode how to involve the validation set so we can save improving checkpoints?

**Comment by voldemortX** [comment]
Unfortunately, we did not leave that functional for lane detection. You will have to do some hacking yourself.

As earlier discussed in #79 . However, in your finetuning case checkpointing may really be important. Perhaps you can add a bit code to save checkpoints every some iterations.

**Comment by MehdiAlilou** [comment]
I actually did some changes to save every couple of epochs, just wanted to double-check with you. I can work on that.

**Comment by voldemortX** [comment]
This issue seems to be addressed. I'll close it for now. Feel free to continue commenting for reopen/open a new one.
